hooks/pre/30-upgrade.rb

```diff
@@ -60,15 +68,20 @@ def fail_and_exit(message)
   upgrade_step :stop_services
   upgrade_step :start_databases
 
-  if Kafo::Helpers.module_enabled?(@kafo, 'katello') || @kafo.param('capsule', 'pulp').value
+  if Kafo::Helpers.module_enabled?(@kafo, 'katello') || @kafo.param('foreman_proxy_plugin_pulp', 'pulpnode_enabled').value
     upgrade_step :migrate_pulp
     upgrade_step :start_httpd
   end
 
+  if @kafo.param('foreman_proxy_plugin_pulp', 'pulpnode_enabled').value
+    upgrade_step :remove_nodes_importers
```

Comment:
I know, I'm suggesting to duplicate migrate_pulp and start_httpd so you only have 2 if statements. I'm ok if you want to leave it the way it is, I was just thinking it might be a little more readable.


```
if capsule
  migrate pulp
  start httpd
  remove nodes importers 

if katello
 migrate pulp
 start httpd
 migrate candlepin
 migrate foreman
 migrate gutterball
 remove nodes distributors
```